astropy/coordinates/representation/geodetic.py

```diff
@@ -10,6 +11,37 @@
 from .cartesian import CartesianRepresentation
 
 
+class WestLongitudeMixin:
+    def to_cartesian(self):
+        self.lon = self.lon.value
```

Comment:
I'm confused - you definitely should not change an internal attribute! How about just
```
cartesian = super().to_cartesian()
np.negative(cartesian._y, out=cartesian._y)
return cartesian
```

Reply:
I was stuck into transforming angular coordinates... now this fixes the test and avoids a double conversion... Thanks!